library/src/main/java/com/digi/xbee/api/models/IP32BitAddress.java

```diff
@@ -0,0 +1,178 @@
+	
+	// Variables
+	private byte[] address;
+	
+	/**
+	 * Class constructor. Instantiates a new object of type 
+	 * {@code IP32BitAddress} with the given parameters.
+	 * 
+	 * @param address The 32 bit IP address as byte array.
+	 * 
+	 * @throws IllegalArgumentException If {@code address.length > 4}.
+	 * @throws NullPointerException If {@code address == null}.
+	 */
+	public IP32BitAddress(byte[] address) {
+		if (address == null)
+			throw new NullPointerException(ERROR_IP_NULL);
+		if (address.length > 4)
+			throw new IllegalArgumentException(ERROR_IP_TOO_LONG);
+		
+		generateIPAddress(address);
+	}
+	
+	/**
+	 * Class constructor. Instantiates a new object of type 
+	 * {@code IP32BitAddress} with the given parameters.
+	 * 
+	 * @param address The 32 bit IP address as integer array.
+	 * 
+	 * @throws IllegalArgumentException If {@code address.length > 8}.
+	 * @throws NullPointerException If {@code address == null}.
+	 */
+	public IP32BitAddress(int[] address) {
+		if (address == null)
+			throw new NullPointerException(ERROR_IP_NULL);
+		if (address.length > 4)
+			throw new IllegalArgumentException(ERROR_IP_TOO_LONG);
+		
+		byte[] byteAddress = new byte[address.length];
+		for (int i = 0; i < address.length; i++)
+			byteAddress[i] = (byte)address[i];
+		
+		generateIPAddress(byteAddress);
+	}
+	
+	/**
+	 * Class constructor. Instantiates a new object of type 
+	 * {@code IP32BitAddress} with the given parameters.
+	 * 
+	 * @param address The 32 bit IP address as string.
+	 * 
+	 * @throws IllegalArgumentException If the given address doesn't match the
+	 *                                  IP address pattern.
+	 * @throws NullPointerException If {@code address == null}.
+	 */
+	public IP32BitAddress(String address) {
+		if (address == null)
+			throw new NullPointerException(ERROR_IP_NULL);
+		
+		Pattern p = Pattern.compile(IP_V4_PATTERN);
+		Matcher m = p.matcher(address);
+		if (!m.matches())
+			throw new IllegalArgumentException(ERROR_IP_INVALID);
+		
+		byte[] byteAddress = new byte[4];
+		
+		byteAddress[0] = (byte)Integer.parseInt(m.group(1));
+		byteAddress[1] = (byte)Integer.parseInt(m.group(2));
+		byteAddress[2] = (byte)Integer.parseInt(m.group(3));
+		byteAddress[3] = (byte)Integer.parseInt(m.group(4));
+		
+		generateIPAddress(byteAddress);
+	}
+	
+	/**
+	 * Generates and saves the IP byte address based on the given byte array.
+	 * 
+	 * @param byteAddress The byte array used to generate the final IP byte 
+	 *                    address.
+	 */
+	private void generateIPAddress(byte[] byteAddress) {
+		this.address = new byte[4];
+		
+		int diff = 4 - byteAddress.length;
+		for (int i = 0; i < diff; i++)
+			this.address[i] = 0;
+		for (int i = diff; i < 4; i++)
+			this.address[i] = byteAddress[i - diff];
+	}
+	
+	/**
+	 * Retrieves the 32 bit IP address value as byte array.
+	 * 
+	 * @return 32 bit IP address value as byte array.
+	 */
+	public byte[] getValue() {
+		return address;
+	}
+	
```

Comment:
I would create a getter that returns a `String` and call it from the `toString()` method (or the other way around).